Library/Homebrew/formula.rb

```diff
@@ -166,6 +166,10 @@ def initialize(name, path, spec)
     validate_attributes!
     @build = active_spec.build
     @pin = FormulaPin.new(self)
+
+    if head && head_pkg_version = latest_head_pkg_version
+      head.version.update_version(head_pkg_version.version)
+    end
```

Comment:
This is definitely incorrect. You should never update head commit to the `latest_head_pkg_version` unconditionaly, because it's quite possible that current active version is not `latest_head_pkg_version`.